framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -1084,7 +1084,7 @@ protected boolean visitWildcardSubtype(
             }
         }
 
-        if (subtype.getUnderlyingType() == superTypeMirror) {
+        if (superTypeMirror.getKind() == TypeKind.WILDCARD) {
```

Comment:
Can you update the documentation in the branch to reflect this change?
What changed in 2.3.2 that makes this the right fix?
Some such comments might make the logic better to follow.